api_tests/lib/cnd/cnd_instance.ts

```diff
@@ -74,7 +75,11 @@ export class CndInstance {
             }
         });
 
-        await sleep(1000); // allow the nodes to start up
+        const logReader = new LogReader(logFile);
+        await logReader.waitForLogMessage("Starting HTTP server on");
+
+        // we emit the log _before_ we start the http server, let's make sure it actually starts up
+        await sleep(1000);
```

Comment:
Can we put less than 1 sec?  Shall we update cnd to log actual readiness?

Reply:
I would leave it with 1 sec with this PR as this is an attempt to fix flaky e2e tests. But I like your proposal to print _actual readiness_. 
@thomaseizinger : can you create a follow-up ticket for that please?

Reply:
I tried but it is not really possible because the future that warp gives us never resolves.

This is just an attempt to give warp _some_ time to actually bind the port etc.
We are waiting for a log line further up already.

A second is probably an overkill but it is hard to say, how much is enough.

You are obviously free to give it a try yourself :)
Not sure if its worth a ticket.
Once all tests run in parallel, this should be barely noticeable.